paint/data/dataset_splits.py

```diff
@@ -0,0 +1,319 @@
+                f"selected split type {split_type} is not supported!"
+            )
+
+        # Calculate the minimum number of possible images as training size + 2 * validation size, since we want at least
+        # as many test images as validation images.
+        minimum_number_of_images = training_size + 2 * validation_size
+
+        # Identify heliostats that do not meet the minimum image requirement.
+        heliostat_image_count = self.metadata.groupby(mappings.HELIOSTAT_ID).size()
+        dropped_heliostats = heliostat_image_count[
+            heliostat_image_count < minimum_number_of_images
+        ].index.tolist()
+
+        # Remove the heliostats not meeting the minimum image requirement.
+        heliostat_split_data = self.metadata.groupby(mappings.HELIOSTAT_ID).filter(
+            lambda x: len(x) >= minimum_number_of_images
+        )
+
+        log.info(
+            f"The following {len(dropped_heliostats)} heliostats have been dropped for not meeting the requirements of {training_size} training "
+            f"images, {validation_size} validation images, and at least {validation_size} test images:"
+        )
+        log.info(dropped_heliostats)
+
+        if len(heliostat_split_data) == 0:
+            raise ValueError(
+                "There are no heliostats left! Your requirements for training and validation size are too"
+                "high. Please retry with a smaller number of required training or validation images!"
+            )
+
+        if split_type == mappings.AZIMUTH_SPLIT:
+            log.info("Preparing azimuth split...")
+            log.info(
+                f"Training Size: {training_size}, Validation Size: {validation_size}, and Test Size > {validation_size}"
+            )
+            heliostat_split_data = heliostat_split_data.groupby(
+                mappings.HELIOSTAT_ID, group_keys=False
+            ).apply(
+                lambda heliostat_data: self._get_azimuth_splits(
+                    heliostat_data=heliostat_data,
+                    training_size=training_size,
+                    validation_size=validation_size,
+                )
+            )
+            log.info("Azimuth split complete!")
+        elif split_type == mappings.SOLSTICE_SPLIT:
+            log.info("Preparing solstice split...")
+            log.info(
+                f"Training Size: {training_size}, Validation Size: {validation_size}, and Test Size > {validation_size}"
+            )
+            heliostat_split_data = heliostat_split_data.groupby(
+                mappings.HELIOSTAT_ID, group_keys=False
+            ).apply(
+                lambda heliostat_data: self._get_solstice_splits(
+                    heliostat_data=heliostat_data,
+                    training_size=training_size,
+                    validation_size=validation_size,
+                )
+            )
+            log.info("Solstice split complete!")
+        else:
```

Comment:
Maybe we can add a random string to test here so we also cover the exception in the else block?

Reply:
We can never get there, because the ``allowed_strings`` are already tested above. So random strings should be caught earlier. Therefore I think we can actually just remove this else block since it will never occur....